store/src/java/com/zimbra/cs/service/mail/Search.java

```diff
@@ -80,6 +80,11 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
         OperationContext octxt = getOperationContext(zsc, context);
         fixBooleanRecipients(request);
         SearchRequest req = zsc.elementToJaxb(request);
+        // ZBUG-1765 ... task selection time date validation is not required
+        if ( req.getSearchTypes().equalsIgnoreCase("task") && req.getCursor() != null) {
```

Comment:
if the functionality is working fine in modern ui then why we need to put a fix in the API layer? did we analyse if modern ui and classic ui are using different request parameters?

Reply:
Yes we are using different request parameters like offset  and limit . We are taking different parameters like in modern ui we are taking limit as 1000 so at a time 1000 tasks will come with a scrollbar
and for classic ui we are using pagination so for each page limit we are taking 50.. 1st time offset will be 0 and limit 50 then in next page offset will be 50 limit 50 like this we are retrieving 50 tasks at a time  … So limit and offset parameter is different in both ui

Reply:
ok thanks for info
so as per changes, we don't support use of cursor in tasks Search request, any reason for this deviation? do we support cursor in mails Search request?
What would be implication of not supporting it and can we get documentation updated for this change?

Reply:
I don’t have idea why that validation was there but I can tell we were getting current date in cursor object and we were comparing it in a query to get data. So we were not facing issue on future date but facing issue on past dates . As my view I have checked there is no use of dates on task labels .

Wherever we have date or time validation we need cursor object with particular date value that we can compare it in queries .

Ya I can provide but in which document I will update ? Or should I create a new document and provide?